### PR TITLE
show # of samples and add link for samples if available

### LINKED ISSUE #970: hierCluster sample selection minor fix
1. at `List samples` option, show number of samples as `List N samples`
2. in sample table, make sample link if available

### PR DESCRIPTION
**Pull request by congyu-lu**
## Description
fix #970 

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/plots/hierCluster.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ class HierCluster extends Matrix {
 				}
 			},
 			{
-				label: 'List samples',
+				label: `List ${clickedSampleNames.length} samples`,
```
Comment (xzhou82) [inline client/plots/hierCluster.js:83]: can it show `Case` for gdc based on dataset config?

File: client/plots/hierCluster.js [hunk 2]
```diff
@@ -190,10 +190,17 @@ class HierCluster extends Matrix {
...
-		const rows =
-			this.app.vocabApi.vocab?.dslabel == 'GDC'
-				? clickedSampleNames.map(c => this.hierClusterData.sampleNameMap[c]).map(c => [{ value: c }])
-				: clickedSampleNames.map(c => [{ value: c }])
+		const templates = self.state.termdbConfig.urlTemplates
+		const rows = templates?.sample
+			? clickedSampleNames.map(c => [{ value: c, url: `${templates.sample.base}${c}` }])
+			: clickedSampleNames.map(c => [{ value: c }])
+
+		if (this.app.vocabApi.vocab?.dslabel == 'GDC') {
```
Comment (xzhou82) [inline client/plots/hierCluster.js:198]: why this hardcode?

Comment (congyu-lu) [reply client/plots/hierCluster.js:198]: only for GDC, we need to convert uuid back to submitter ID

**Review by xzhou82 — APPROVED** [review]
thanks. we can fix the hardcode in the big refactor

**Pull request merged**